File: src/WellTool.Core/hutool-core/src/main/java/cn/hutool/core/convert/impl/TimeZoneConverter.java
package cn.hutool.core.convert.impl;

import cn.hutool.core.convert.AbstractConverter;
import cn.hutool.core.date.TimeZoneUtil;

import java.util.TimeZone;

/**
 * TimeZone转换器
 * @author Looly
 *
 */
public class TimeZoneConverter extends AbstractConverter<TimeZone>{
	private static final long serialVersionUID = 1L;

	@Override
	protected TimeZone convertInternal(Object value) {
		return TimeZoneUtil.getTimeZone(convertToStr(value));
	}

}

File: src/WellTool.Core/Convert/impl/TimeZoneConverter.cs
using System;

namespace WellTool.Core.Convert.Impl;

/// <summary>
/// TimeZone杞崲鍣?
/// </summary>
public class TimeZoneConverter : AbstractConverter<TimeZoneInfo>
{
    protected override TimeZoneInfo ConvertInternal(object value)
    {
        var str = ConvertToStr(value);
        try
        {
            return TimeZoneInfo.FindSystemTimeZoneById(str);
        }
        catch
        {
            return TimeZoneInfo.Local;
        }
    }
}

